bin/yarn

```diff
@@ -1,16 +1,15 @@
 #!/usr/bin/env ruby
-require 'pathname'
 
-APP_ROOT = File.expand_path('..', __dir__)
+APP_ROOT = File.expand_path("..", __dir__)
 Dir.chdir(APP_ROOT) do
-  executable_path = ENV["PATH"].split(File::PATH_SEPARATOR).find do |path|
-    normalized_path = File.expand_path(path)
+  yarn = ENV["PATH"].split(File::PATH_SEPARATOR).
+    select { |dir| File.expand_path(dir) != __dir__ }.
+    product(["yarn", "yarnpkg", "yarn.cmd", "yarn.ps1"]).
+    map { |dir, file| File.expand_path(file, dir) }.
+    find { |file| File.executable?(file) }
 
-    normalized_path != __dir__ && File.executable?(Pathname.new(normalized_path).join('yarn'))
-  end
-
-  if executable_path
-    exec File.expand_path(Pathname.new(executable_path).join('yarn')), *ARGV
+  if yarn
+    exec yarn, *ARGV
   else
     $stderr.puts "Yarn executable was not detected in the system."
```

Comment:
I think it would be good to print the yarn path, if any, to help debug any issue.

Reply:
This is a very simple binstub copied over by shakapacker (see the [source file](https://github.com/shakacode/shakapacker/tree/master/lib/install/bin/yarn)). 

You should be good to use it if you have yarn installed as [mentioned](https://github.com/DFE-Digital/early-careers-framework#prerequisites) in the README. If not, it will ask you to download it.